### PR TITLE
Add GuildUser endpoints.

### PR DESCRIPTION
**Pull request by jasonhaxstuff**

### CODE REVIEW

File: src/services/guild.ts [hunk 1]
```diff
@@ -107,4 +108,31 @@ export class GuildService implements BaseService<Guild> {
...
+
+    if (!user) {
+      return
+    }
+
+    this.userRepository.remove(user)
+  }
+
+  public async updateUser (id: string, userId: number, user: GuildUser) {
+    return this.userRepository.update({ guild: { id }, id: userId }, user)
```
Comment (tannerz28) [inline src/services/guild.ts]: `id: userId` needs to be changed to `user: { id: userId }`

File: src/services/guild.ts [hunk 2]
```diff
@@ -107,4 +108,31 @@ export class GuildService implements BaseService<Guild> {
...
+    const user = await this.userRepository.findOne({ where: { guild: { id }, id: { userId } } })
+
+    if (!user) {
+      return
+    }
+
+    this.userRepository.remove(user)
+  }
+
+  public async updateUser (id: string, userId: number, user: GuildUser) {
```
Comment (tannerz28) [inline src/services/guild.ts]: `userId` should be a string.

File: src/controllers/guild.ts [hunk 3]
```diff
@@ -335,4 +335,99 @@ export class GuildController implements BaseController<Guild, string> {
...
+
+  /**
+   * Gets a Guild user by ID.
+   *
+   * GET /:id/users/:userId
+   * @param request
+   * @param response
+   */
+  @httpGet('/:id/users/:userId')
+  async getUserById (@requestParam('id') id: string, @requestParam('userId') userId: number) {
```
Comment (tannerz28) [inline src/controllers/guild.ts]: `userId` should be a string.

File: src/controllers/guild.ts [hunk 4]
```diff
@@ -335,4 +335,99 @@ export class GuildController implements BaseController<Guild, string> {
...
+
+  /**
+   * Updates a Guild user by ID.
+   *
+   * PUT /:id/users/:userId
+   * @param request
+   * @param response
+   */
+  @httpPut('/:id/users/:userId')
+  async updateUserById (@requestParam('id') id: string, @requestParam('userId') userId: number, @request() request: Request) {
```
Comment (tannerz28) [inline src/controllers/guild.ts]: `userId` should be a string.

File: src/controllers/guild.ts [hunk 5]
```diff
@@ -335,4 +335,99 @@ export class GuildController implements BaseController<Guild, string> {
...
+
+  /**
+   * Deletes a Guild user by ID.
+   *
+   * DELETE /:id/users/:userId
+   * @param request
+   * @param response
+   */
+  @httpDelete('/:id/users/:userId')
+  async deleteUserById (@requestParam('id') id: string, @requestParam('userId') userId: number, @request() request: Request) {
```
Comment (tannerz28) [inline src/controllers/guild.ts]: `userId` should be a string. Also remove the third parameter, `@request() request: Request`. Dont need that in a Delete request.

File: src/controllers/guild.ts [hunk 6]
```diff
@@ -335,4 +335,99 @@ export class GuildController implements BaseController<Guild, string> {
...
+    const deleteResponse = this.guildService.deleteUser(id, userId)
+    await deleteResponse
+      .then(() => {
+        this.socketService.send(Events.guild.user.deleted, {
+          guildId: id,
+          userId: request.body
```
Comment (tannerz28) [inline src/controllers/guild.ts]: This line should just be `userId,`

File: src/services/guild.ts [hunk 7]
```diff
@@ -107,4 +108,31 @@ export class GuildService implements BaseService<Guild> {
   public async updateSettings (id: string, settings: GuildSettings) {
     return this.settingsRepository.update({ guild: { id } }, settings)
   }
+
+  public async getUsers (id: string) {
+    return this.userRepository.find({ where: { guild: { id } } })
+  }
+
+  public getUserById (id: string, userId: number) {
```
Comment (tannerz28) [inline src/services/guild.ts]: `userId` should be a number.

File: src/services/guild.ts [hunk 8]
```diff
@@ -107,4 +108,31 @@ export class GuildService implements BaseService<Guild> {
...
+    return this.userRepository.findOne({ where: { guild: { id }, id: { userId } } })
```
Comment (tannerz28) [inline src/services/guild.ts]: `id: { userId }` should be `user: { id: userId }`

File: src/services/guild.ts [hunk 9]
```diff
@@ -107,4 +108,31 @@ export class GuildService implements BaseService<Guild> {
...
+  }
+
+  public async deleteUser (id: string, userId: number) {
+    const user = await this.userRepository.findOne({ where: { guild: { id }, id: { userId } } })
+
+    if (!user) {
+      return
+    }
+
+    this.userRepository.remove(user)
```
Comment (tannerz28) [inline src/services/guild.ts]: add `return` to the beginning of this line.

File: src/services/guild.ts [hunk 10]
```diff
@@ -107,4 +108,31 @@ export class GuildService implements BaseService<Guild> {
...
+  public getUserById (id: string, userId: number) {
+    return this.userRepository.findOne({ where: { guild: { id }, id: { userId } } })
+  }
+
+  public async createUser (id: string, user: GuildUser) {
+    user.dateJoined = new Date()
+    return this.userRepository.save(user)
+  }
+
+  public async deleteUser (id: string, userId: number) {
```
Comment (tannerz28) [inline src/services/guild.ts]: `userId` should be a string.

File: src/services/guild.ts [hunk 11]
```diff
@@ -107,4 +108,31 @@ export class GuildService implements BaseService<Guild> {
...
+    return this.userRepository.findOne({ where: { guild: { id }, user: { id: userId } } })
+  }
+
+  public async createUser (id: string, user: GuildUser) {
+    user.dateJoined = new Date()
+    return this.userRepository.save(user)
+  }
+
+  public async deleteUser (id: string, userId: string) {
+    const user = await this.userRepository.findOne({ where: { guild: { id }, id: { userId } } })
```
Comment (tannerz28) [inline src/services/guild.ts]: `id: { userId }` should be `user: { id: userId }`

**Comment by jasonhaxstuff** [comment]
@tannerz28 done * 2

**Pull request merged**